packages/typespec-client-generator-core/src/clients.ts

```diff
@@ -133,7 +133,9 @@ function getSdkEndpointParameter<TServiceOperation extends SdkServiceOperation =
 ): [SdkEndpointParameter, readonly Diagnostic[]] {
   const diagnostics = createDiagnosticCollector();
   const rawClient = client.__raw;
-  const servers = getServers(context.program, client.__raw.service);
+  // For multiple services, just take the first one to get servers
+  const service = Array.isArray(rawClient.service) ? rawClient.service[0] : rawClient.service;
+  const servers = getServers(context.program, service);
```

Comment:
I think we should report something if we are taking the first ignoring others.
I am not sure I know what the "servers" here represent, but if we are ignoring others, we should report something when we ignore them and our user could rethink.